### PR TITLE
Change method signature of ManifestReader::read (#459)

### PR DESCRIPTION
**Pull request by lxynov**
This commit is part of the efforts to decouple TableMetadataParser from
TableOperations.

#459

### CODE REVIEW

**Comment by rdblue** [comment]
Thanks @linxingyuan1102! I'm out until Wednesday next week, but I'll take a look at this and #459 when I'm back.

**Comment by rdblue** [comment]
This looks reasonable to me. @aokolnychyi, any comments?

**Comment by aokolnychyi** [comment]
I am +1. I think it is in line with #511 and will eliminate the need to build a spec lookup function in utilities.

**Comment by linxingyuan1102** [comment]
@rdblue @aokolnychyi Thank you for the review! I'll continue working on steps 2&3 listed in #459 after this PR is merged.

**Comment by aokolnychyi** [comment]
I think we can merge this PR now. @rdblue, what do you think?

**Comment by rdblue** [comment]
Sounds good to me.

File: core/src/main/java/org/apache/iceberg/DataTableScan.java [hunk 1]
```diff
@@ -83,7 +83,8 @@ protected TableScan newRefinedScan(
     Iterable<CloseableIterable<FileScanTask>> readers = Iterables.transform(
         matchingManifests,
         manifest -> {
-          ManifestReader reader = ManifestReader.read(ops.io().newInputFile(manifest.path()), ops.current()::spec);
+          ManifestReader reader = ManifestReader.read(ops.io().newInputFile(manifest.path()),
```
Comment (aokolnychyi) [inline core/src/main/java/org/apache/iceberg/DataTableScan.java]: nit: would it be more readable if we place each argument on a separate line?

```
ManifestReader.read(
    ops.io().newInputFile(manifest.path()),
    ops.current().specsById());
```

**Comment by lxynov** [comment]
@aokolnychyi Sure, updated.

**Comment by aokolnychyi** [comment]
@lxynov, thanks! I kept the issue open as this PR is the first step.

**Pull request merged**